File: app/src/main/java/com/xmx/tango/module/import_/ImportUtil.java
package com.xmx.tango.module.import_;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.support.v7.app.AlertDialog;

import com.xiaoleilu.hutool.lang.Conver;
import com.xmx.tango.module.tango.Tango;
import com.xmx.tango.utils.StrUtil;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Created by The_onE on 2017/7/28.
 */

public class ImportUtil {
    public static void showDialog(List<String> dialogStrings,
                                  final ArrayList<String> intentStrings,
//                                  final ArrayList<Tango> tangoList,
                                  final String type,
                                  final Context context) {
        String array[] = new String[dialogStrings.size()];
        array = dialogStrings.toArray(array);
        new AlertDialog.Builder(context)
                .setTitle("识别出的単語")
                .setItems(array, null)
                .setPositiveButton("导入", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        StrUtil.INSTANCE.showToast(context, "正在导入，请稍后");
//                            new NewThread() {
//                                @Override
//                                public void process() {
//                                    for (Tango t : tangoList) {
//                                        TangoEntityManager.getInstance().insertData(t);
//                                    }
//                                    showToast("导入成功");
//                                    EventBus.getDefault().post(new TangoListChangeEvent());
//                                }
//                            }.start();
//                        Intent service = new Intent(context, ImportService.class);
//                        service.putStringArrayListExtra("list", intentStrings);
                        Intent service = new Intent(context, ImportEntityService.class);
                        ArrayList<Tango> tangoList = new ArrayList<>();
                        for (String str : intentStrings) {
                            String[] strings = str.split(",");
                            tangoList.add(makeTango(strings, type));
                        }
                        service.putParcelableArrayListExtra("list", tangoList);
                        service.putExtra("type", type);
                        context.startService(service);
                    }
                })
                .setNegativeButton("取消", null)
                .show();
    }

    public static Tango makeTango(String[] strings, String type) {
        Tango tango = new Tango();
        try {
            tango.setWriting(strings[0]);
            tango.setPronunciation(strings[1]);
            tango.setMeaning(strings[2]);
            tango.setTone(Conver.toInt(strings[3], -1));
            tango.setPartOfSpeech(strings[4]);
            tango.setImage(strings[5]);
            tango.setVoice(strings[6]);
            tango.setScore(Conver.toInt(strings[7], 0));
            tango.setFrequency(Conver.toInt(strings[8], 0));
            tango.setAddTime(new Date(Conver.toLong(strings[9], 0L)));
            tango.setLastTime(new Date(Conver.toLong(strings[10], 0L)));
            tango.setFlags(strings[11]);
            tango.setDelFlag(Conver.toInt(strings[12], 0));
            tango.setType(strings[13]);
        } catch (IndexOutOfBoundsException e) {
        } finally {
            if (!StrUtil.INSTANCE.isBlank(type)) {
                tango.setType(type);
            }
            if (tango.getAddTime().getTime() == 0) {
                tango.setAddTime(new Date());
            }
        }

        return tango;
    }
}

File: app/src/main/java/com/xmx/tango/module/imp/ImportUtil.kt
package com.xmx.tango.module.imp

import android.content.Context
import android.content.Intent
import android.support.v7.app.AlertDialog

import com.xiaoleilu.hutool.lang.Conver
import com.xmx.tango.module.tango.Tango
import com.xmx.tango.utils.CsvUtil
import com.xmx.tango.utils.StrUtil
import org.apache.commons.csv.CSVRecord

import java.util.Date

/**
 * Created by The_onE on 2017/7/28.
 * 导入单语工具类
 */
object ImportUtil {
    /**
     * 显示识别出单语的对话框，点击确认调用Service进行导入
     * @param dialogStrings 在对话框中显示给用户的字符串数组
     * @param path 要导入的CSV文件路径
     * @param type 导入单语统一设置的类型，若为null则保持各单语原类型
     * @param context 当前上下文
     */
    fun showFileDialog(dialogStrings: List<String>,
                       path: String,
                       type: String,
                       context: Context) {
        val array = dialogStrings.toTypedArray()
        AlertDialog.Builder(context)
                .setTitle("识别出的単語")
                .setItems(array, null)
                .setPositiveButton("导入") { _, _ ->
                    StrUtil.showToast(context, "正在导入，请稍后")
                    val service = Intent(context, ImportFileService::class.java)
                    service.putExtra("path", path)
                    service.putExtra("type", type)
                    context.startService(service)
                }
                .setNegativeButton("取消", null)
                .show()
    }

    /**
     * 将CSV记录转化为单语实体
     * @param record 一条CSV记录
     * @param type 单语统一设置的类型，若为null则保持各单语原类型
     * @return 单语实体
     */
    private fun convertTango(record: CSVRecord, type: String): Tango {
        val tango = Tango()
        try {
            tango.writing = record[CsvUtil.WRITING]
            tango.pronunciation = record[CsvUtil.PRONUNCIATION]
            tango.meaning = record[CsvUtil.MEANING]
            tango.tone = Conver.toInt(record[CsvUtil.TONE], -1)
            tango.partOfSpeech = record[CsvUtil.PART_OF_SPEECH]
            tango.image = record[CsvUtil.IMAGE]
            tango.voice = record[CsvUtil.VOICE]
            tango.score = Conver.toInt(record[CsvUtil.SCORE], 0)
            tango.frequency = Conver.toInt(record[CsvUtil.FREQUENCY], 0)
            tango.addTime = Date(Conver.toLong(record[CsvUtil.ADD_TIME], 0L))
            tango.lastTime = Date(Conver.toLong(record[CsvUtil.LAST_TIME], 0L))
            tango.flags = record[CsvUtil.FLAGS]
            tango.delFlag = Conver.toInt(record[CsvUtil.DEL_FLAG], 0)
            tango.type = record[CsvUtil.TYPE]
        } catch (e: IndexOutOfBoundsException) {
        } finally {
            if (!StrUtil.isBlank(type)) {
                tango.type = type
            }
            if (tango.addTime.time == 0L) {
                tango.addTime = Date()
            }
        }

        return tango
    }

    /**
     * 将CSV记录列表转化为单语实体列表
     * @param list CSV记录列表
     * @param type 单语统一设置的类型，若为null则保持各单语原类型
     * @return 单语实体列表
     */
    fun convertTangoList(list: List<CSVRecord>, type: String): List<Tango> =
            list.map { convertTango(it, type) }
}
